require.js

```diff
@@ -592,6 +593,33 @@
                             Promise.all(promises)) : null;
             }, function (error) {
                 getModuleDescriptor(topId).error = error;
+            }).then(function (value) {
+                var module = getModuleDescriptor(topId);
+                if (module.location && (endsWith(module.location, ".meta") || endsWith(module.location, ".mjson"))) {
+                    if (typeof module.exports !== "object" && typeof module.text === "string") {
+                        if (Require.delegate && typeof Require.delegate.compileMJSONFile === "function") {
+                            return Require.delegate.compileMJSONFile(
+                                module.text, requireForId(module.id), module.id
+                            ).then(function (root) {
+                                module.exports = JSON.parse(module.text);
+                                if (module.exports.montageObject) {
+                                    throw new Error(
+                                        'using reserved word as property name, \'montageObject\' at: ' +
+                                        module.location
+                                    );
+                                }
+
+                                module.exports.montageObject = root;
+                                return value;
+                            });
+                        } else {
+                            module.exports = JSON.parse(module.text);
+                        }
+                    }
+                }
+                return value;
+            },function (error) {
+                getModuleDescriptor(topId).error = error;
```

Comment:
This whole section looks like it should be a compiler. `deepLoad` transitively calls `config.compile`.

Reply:
It can't because deserialization has today 2 async steps, mr isn't built to deal with that, there's async loading of dependencies, then sync phases only. Once we can deserialize synchronously, we can remove that here and do otherwise.

Reply:
I'm not understanding, what are the two async steps? Loading the mjson file then loading the modules referenced by the mjson file? That all sounds like one async step, which is what `deepLoad` does. Is the second step just because we don't currently have a sync mode for the deserializer? If so, that's being done in montagejs/montage#2011, and that PR is ready to go after your review IMO.

Reply:
Yes, that's exactly why, ok, let's take care of montagejs/montage#2011 and then benefit from it here